### DISCUSSION TITLE
concurrency报错，显示下载chunk失序，但在terminal中测试正常

### DISCUSSION THREAD

**Discussion by YuTheon** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#66] [Q&A]
在pytest中报错
```
>  assert th.strip() == received_hash_str.strip(), f"received data mismatch, expect hash: {target_hash}, actual: {received_hash_str}"
E   AssertionError: received data mismatch, expect hash: ['45acace8e984465459c893197e593c36daf653db', '3b68110847941b84e8d05417a5b2609122a56314'], actual: 18297c48b955408c01c72d95764476710425fb7b

考虑是接收到的pkt失序装配，但是在terminal中运行之后收到的包正常，请求帮助qwq

**Comment by TANSixu** [comment]
Hi, 请问在terminal中运行时是否有正常启动模拟器呢，以及 错误下载的数据算出来的hash是否每次都相同呢

**Reply by YuTheon** [reply]
有正常启动模拟器，错误下载的数据哈希值是相同的。虽然还不知道terminal中为什么没有报错，但是找到在pytest中错误出现的原因，是因为一个peer的chunk先传完时，另一个peer会少接收一个pkt

**Reply by TANSixu** [reply]
> 有正常启动模拟器，错误下载的数据哈希值是相同的。虽然还不知道terminal中为什么没有报错，但是找到在pytest中错误出现的原因，是因为一个peer的chunk先传完时，另一个peer会少接收一个pkt

可以检查一下是否在第二个chunk还没下载完时peer就写入了文件，因为测试脚本在检测到result文件生成后就会停止所有peer。

**Reply by YuTheon** [reply]
哦对！原来如此，这里是每次接收一个完整chunk就写入文件了，谢谢学长！
